Lesson18/deniz/script.js

```diff
@@ -0,0 +1,192 @@
+// [1, 'apple', true, null]
+const address = {
+  line1: 'Kungstorget 1',
+  line2: null,
+  city: 'Gothenburg',
+  postalCode: '411 17',
+  country: 'Sweden'
+}; 
+
+console.log(address.country);
+
+// 1. Write a function that takes an array of integers and returns the sum of all elements.
+// Input: [1, 2, 3, 4]
+// Output: 10
+const integersEx1 = [1, 2, 3, 4];
+const integersEx1V2 = [10, 2, 30, 4];
+
+function getSumOfArray (numbers) {
+  // check if numbers is actually an array and only contains numbers
+  let result = numbers[0];
+  for (let i = 1; i < numbers.length; i++) {
+    result += numbers[i];
+  }
+  return result;
+}
+
+console.log("Exercise 1: ", getSumOfArray(integersEx1V2), getSumOfArray(integersEx1), getSumOfArray([10, 10, 10]));
+console.log('----------------------------');
+// 2. Write a function that takes an array of integers and returns the average of all elements.
+//Input: [1, 2, 3, 4]
+// Output: 2.5
+const integersEx2 = [1, 2, 3, 4];
+
+ function getAvarege (numbers) {
+  return getSumOfArray(numbers) / numbers.length;
+ }
+
+console.log("Exercise 2:", getAvarege(integersEx2));
+console.log('----------------------------');
+// 3. Write a function that takes an array of strings and concatenates them together adding spaces.
+// Input: ["Hello", "World", "!"]
+// Output: "Hello World !"
+
+let stringE3 = ["Hello", "World", "!"];
+
+function concatenateThemTogether (arr) {
+  return arr.join(" ");
+}
+
+console.log("Exercise 3:", concatenateThemTogether(stringE3));
+
+console.log('----------------------------');
+// 4. Write a function that takes an array of integers and returns a new array with odd numbers only.
+// Input: [1, 2, 3, 4, 7, 10]
+// Output: [1, 3, 7]
+let arrayEx = [1, 2, 3, 4, 7, 10];
+
+function arryWithOddNumbers (numbers) {
```

Comment:
A 'getOddNumbers' could be a more descriptive name. Try to use a verb as the first word in the name of a function